src/components/Navbar/NavItem.js

```diff
@@ -0,0 +1,29 @@
+//Dependencies
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+//Components, assets, actions, styles etc..
+import { setActiveMenu } from '../../actions/index';
+
+const NavItem = (props) => {
+  const navState = useSelector((state) => state.nav);
+  const dispatch = useDispatch();
+
+  return (
+    <Link
+      className={props.name === 'Donate' ? 'donate--link' : 'nav--link'}
+      onMouseEnter={() => {
+        dispatch(setActiveMenu(props.name));
+      }}
+      to={props.children ? '#' : props.linkTo}
+    >
+      <li href="#" className={props.name === 'Donate' ? 'donate--link__txt' : 'nav--link__txt'}>
+        {props.name}
+      </li>
+      {navState.activeMenu === props.name && props.children}
+    </Link>
```

Comment:
This logic could use some work this dispatch is always run when mouse enters, what happens where there is no children to show, this might cause a memory leak depending on how react handles on mouse enter

Reply:
I left the dispatch() on links with no children so that if in the future the client needs to add children to a nav-link our navItem component can always look at the redux store and handle the closing of the DropDownMenu when hovering across links.

Reply:
dispatch now only fires if navlink has children